src/localisation/test/ekf_rostest.cpp

```diff
@@ -0,0 +1,274 @@
+/*
+ * Created By: Marcus Swift
+ * Created On: November 20th, 2017
+ * Description: Intergration testing for EKF
+ */
+
+#include <EKF.h>
+#include <gtest/gtest.h>
+#include <random>
+#include <ros/ros.h>
+
+/**
+ * This is the helper class which will publish and subscribe messages which will
+ * test the node being instantiated
+ * It contains at the minimum:
+ *      publisher - publishes the input to the node
+ *      subscriber - publishes the output of the node
+ *      callback function - the callback function which corresponds to the
+ * subscriber
+ *      getter function - to provide a way for gtest to check for equality of
+ * the message recieved
+ */
+class EKFTest : public testing::Test {
+  protected:
+    virtual void SetUp() {
+        // set up publsihers and subscribers
+        test_gps_publisher =
+        nh_.advertise<nav_msgs::Odometry>("/gps_driver/odom", 10);
+        test_encoder_publisher = nh_.advertise<nav_msgs::Odometry>("/odom", 10);
+        test_imu_publisher     = nh_.advertise<sensor_msgs::Imu>("/imu", 10);
+        test_pose_subscriber =
+        nh_.subscribe("/cmd_pose", 10, &EKFTest::poseCallback, this);
+
+        // Let the publishers and subscribers set itself up timely
+        ros::Rate loop_rate(10);
+        loop_rate.sleep();
+    }
+
+    // node handler, publsiher and subscriber definitions
+    ros::NodeHandle nh_;
+    geometry_msgs::Pose pose_data;
+    ros::Publisher test_gps_publisher;
+    ros::Publisher test_encoder_publisher;
+    ros::Publisher test_imu_publisher;
+    ros::Subscriber test_pose_subscriber;
+
+  public:
+    void poseCallback(const geometry_msgs::Pose::ConstPtr pose_msg) {
+        pose_data.position    = pose_msg->position;
+        pose_data.orientation = pose_msg->orientation;
+    }
+};
+
+TEST_F(EKFTest, testEKF) {
+    // set up gaussian noise generator
+    std::random_device rd;
+    std::mt19937 e2(rd());
+    std::normal_distribution<> noise(0, 5);
```

Comment:
I think we should use a constant seed here when creating the `std::mt19937`, otherwise this test is non-deterministic (that is, we could run it multiple times and get different results). I think this *may* be why CI is currently failing this test.